### PR TITLE
Add AWS Foundational Security Best Practices S3 benchmark. Closes #71

### LINKED ISSUE #71: Add AWS Foundational Security Best Practices S3 benchmark
**Is your feature request related to a problem? Please describe.**
I'd like to be able to run theS3 benchmark for the AWS Foundational Security Best Practices framework.

**Describe the solution you'd like**
Implement S3 benchmark..

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [x] Issue(s) linked

### CODE REVIEW

File: query/s3/s3_public_access_block_account.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+select
```
Comment (cbruno10) [inline query/s3/s3_public_access_block_account.sql:1]: @khushboo9024 Would we be able to make the query a bit more robust, like the `s3_public_access_block_bucket_account.sq` query (in particular when all settings are not blocked)?

File: query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+)
+select
+  -- Required Columns
+  a.arn as resource,
+  case
+    when b.arn is null then 'ok'
+    else 'alarm'
+  end as status,
+  case
+    when b.arn is null then title || ' having no across-account access.'
```
Comment (cbruno10) [inline query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:32]:
```suggestion
    when b.arn is null then title || ' restricts cross-account bucket access.'
```

File: query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    else title || ' having across-account access.'
```
Comment (cbruno10) [inline query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:33]:
```suggestion
    else title || ' allows cross-account bucket access.'
```

File: query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  end as reason,
+  -- Additionl Dimensions
+  a.region,
+  a.account_id
+from
+  aws_s3_bucket a
+  left join crossaccount_buckets b on a.arn = b.arn;
```
Comment (cbruno10) [inline query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:40]:
```suggestion
  left join cross_account_buckets b on a.arn = b.arn;
```

File: query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    string_to_array(p, ':') as pa,
+    jsonb_array_elements_text(s -> 'Action') as a
+  where
+    s ->> 'Effect' = 'Allow'
+    and (
+      pa [5] != account_id
+      or p = '*'
+    )
+    and a in (
+      's3:deletebucketpolicy',
```
Comment (cbruno10) [inline query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:17]: Are there other permissions the control cares about? Why were these picked in particular?

File: foundational_security/s3.sp [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  severity      = "high"
+  sql           = query.s3_bucket_policies_restricts_permissions_to_other_aws_account.sql
+  #documentation = file("./foundational_security/docs/foundational_security_s3_6.md")
+
+  tags = merge(local.foundational_security_s3_common_tags, {
+    foundational_security_item_id  = "s3_6"
+    foundational_security_category = "sensitive_api_operations_actions_restricted"
+  })
+}
+
```
Comment (cbruno10) [inline foundational_security/s3.sp:98]: Remove extra newlines

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comments

File: query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql (hunk 5 above)
Comment (khushboo9024) [reply query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:17]: @cbruno10 The control does care about the other permissions, it is as per the requirement of the control.
https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-s3-6

Comment (cbruno10) [reply query/s3/s3_bucket_policies_restricts_permissions_to_other_aws_account.sql:17]: `query/s3/s3_bucket_policy_restricts_cross_account_permission_changes.sql`

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**